### PR TITLE
Exclude method OPTIONS in the list of supported request methods

### PR DESCRIPTION
**Pull request by harlekoy**

### CODE REVIEW

File: src/Listeners/LogRouteUsage.php [hunk 1]
```diff
@@ -33,15 +34,21 @@ protected function shouldLogUsage($event)
...
+        if (isset($regex['name']) && $regex['name'] && preg_match($regex['name'], $name)) {
             return false;
         }
 
         if (isset($regex['uri']) && $regex['uri'] && preg_match($regex['uri'], $route->uri)) {
             return false;
         }
 
+        // Lets exclude OPTIONS in the list of methods supported
+        if (Str::is($event->request->getMethod(), 'OPTIONS')) {
```
Comment (julienbourdeau) [inline src/Listeners/LogRouteUsage.php]: 👍

File: src/Listeners/LogRouteUsage.php [hunk 2]
```diff
@@ -33,15 +34,21 @@ protected function shouldLogUsage($event)
 
         $route = $event->request->route();
         $regex = config('route-usage.excluding-regex');
+        $name = method_exists($route, 'getName') ? $route->getName() : null;
```
Comment (julienbourdeau) [inline src/Listeners/LogRouteUsage.php]: Can you explain when this method would not be defined?

**Comment by julienbourdeau** [comment]
Thanks for your contributions! 👍

**Comment by ingalless** [comment]
Could you remove the dependency on `\Illuminate\Support\Str` by using the "isMethod()" method of the request?
Like so:
```php
<?php $event->request->isMethod('options'); ?>
```
[More info here](https://laravel.com/docs/6.x/requests#accessing-the-request)

**Comment by julienbourdeau** [comment]
Good idea. It's also more readable.

Any reason why you want to remove the dependency to Str (since this package only works with Laravel anyway)?

**Comment by ingalless** [comment]
None in particular, other than it could read a little better as you've said, and as it's on the request object Laravel will already be handling bizarre edge cases (if there are any?)

**Comment by julienbourdeau** [comment]
@ingalless @harlekoy In the end, I went for #12. Thanks for the suggestions and ideas 💯

**Pull request closed without merging**